api/scpca_portal/views/dataset.py

```diff
@@ -49,3 +49,26 @@ def destroy(self, request, pk=None):
             {"detail": "Deleting datasets is not allowed."},
             status=status.HTTP_405_METHOD_NOT_ALLOWED,
         )
+
+    def perform_create(self, serializer):
+        dataset = serializer.save()
+        if dataset.start:
+            self.submit_job(dataset)
+
+    def perform_update(self, serializer):
+        dataset = serializer.save()
+        if dataset.start:
```

Comment:
Before applying the posted data to the instance we should assert that dataset.start is not `True`. Since that is our flag that indicates something is modifiable